File: app/src/main/java/com/android/sonu/ummsplash/ui/home/HomeModule.java
package com.android.sonu.ummsplash.ui.home;

import android.app.Activity;

import com.android.sonu.ummsplash.di.ActivityScoped;
import com.android.sonu.ummsplash.di.FragmentScoped;

import dagger.Binds;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

/**
 * Created by amanshuraikwar on 18/12/17.
 */

@Module
public abstract class HomeModule {

    @FragmentScoped
    @ContributesAndroidInjector
    abstract HomeFragment homeFragment();

    @ActivityScoped
    @Binds
    abstract HomeContract.Presenter getHomePresenter(HomePresenter presenter);

    @ActivityScoped
    @Binds
    abstract Activity activity(HomeActivity activity);
}

File: app/src/main/java/io/github/amanshuraikwar/splash/ui/home/HomeModule.kt
package io.github.amanshuraikwar.splash.ui.main

import android.support.v7.app.AppCompatActivity
import dagger.Binds
import dagger.Module
import io.github.amanshuraikwar.splash.di.ActivityContext
import io.github.amanshuraikwar.splash.di.ActivityScope
import io.github.amanshuraikwar.splash.ui.home.HomeContract
import io.github.amanshuraikwar.splash.ui.home.HomePresenter

/**
 * Dagger Module to provide MainActivity related instances.
 *
 * @author Amanshu Raikwar
 * Created by amanshuraikwar on 30/04/18.
 */
@Module abstract class HomeModule {

    @ActivityScope @Binds
    abstract fun prsntr(presenter: HomePresenter): HomeContract.Presenter

    @ActivityScope @Binds @ActivityContext
    abstract fun activity(activity: AppCompatActivity): AppCompatActivity
}